// @fermyon/spin-sdk/lib/modules/spinSdk.js
import { router } from "./router";
import { utils } from "./utils";
const kv = {
    open: (name) => {
        let store = __internal__.spin_sdk.kv.open(name);
        store.getJson = (key) => {
            return JSON.parse(new TextDecoder().decode(store.get(key)));
        };
        store.setJson = (key, value) => {
            store.set(key, JSON.stringify(value));
        };
        return store;
    },
    openDefault: () => {
        let store = kv.open("default");
        return store;
    }
};
/**  features
 */
/** @deprecated */
const spinSdk = {
    config: __internal__.spin_sdk.config,
    redis: __internal__.spin_sdk.redis,
    kv: kv,
    mysql: __internal__.spin_sdk.mysql,
    pg: __internal__.spin_sdk.pg,
    sqlite: __internal__.spin_sdk.sqlite,
    utils: utils,
    Router: () => {
        return router();
    }
};
const Config = __internal__.spin_sdk.config;
const Redis = __internal__.spin_sdk.redis;
const Kv = kv;
const Mysql = __internal__.spin_sdk.mysql;
const Pg = __internal__.spin_sdk.pg;
const Sqlite = __internal__.spin_sdk.sqlite;
export { spinSdk };
export { Config, Redis, Kv, router, Mysql, Pg, Sqlite };
